### DISCUSSION TITLE
Transport based on fetch()

### DISCUSSION THREAD

**Discussion by Type1J** [sockjs/sockjs-client#561] [Ideas]
I am using SockJS in environments where an embedded JavaScript interpreter is used.

I have implemented fetch() in platform specific exposed primitives, and I'd like SockJS to use that fetch() API.

If fetch is available, which at this point it will be in most browsers, then it should be used instead of an XMLHttpRequest based transport.

**Comment by brycekahle** [comment]
A PR for this would be much appreciated, but is not something that is high on my priority list.

**Comment by izogfif** [comment]
It's 2022, and SockJS is useless when connections are made in Service Workers since the only transport it supports is WebSockets. It's possible to use WebSockets without SockJS in modern day. The main advantage of SockJS is SockJS protocol and fall-backs that allow more or less reliable connection in edge cases where firewalls with DPI (deep packet inspection) / proxies / other third-party software prevent WebSocket connections.
Right now I have to use Shim / Polyfill / whatever this thing is called to emulate `XMLHttpRequest` in Service Worker via `fetch` and `Request` API (https://gist.github.com/izogfif/1886cb4c2a34f41a3db64977605839be). Which is utter nonsense.
